lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -780,6 +780,12 @@ CFEPAnalyzer<FunctionOracle>::collectDirectCFG(OutlinedFunction *F) {
           } else {
             Instruction *I = &(*Succ->begin());
 
+            // Did we meet the end of the cloned basic block?
+            if (isa<ReturnInst>(I)) {
+              revng_assert(Succ->getInstList().size() == 1);
+              continue;
+            }
+
             if (auto *Call = getCallTo(I, PreHookMarker.F)) {
```

Comment:
```suggestion
              // Do nothing
            } else if (auto *Call = getCallTo(I, PreHookMarker.F)) {
```

Reply:
IMHO `continue` here makes the code more readable (if it's OK with you).

Reply:
Nah. `continue` introduces (in this case unnecessary) non-obvious control-flow. What you're handling is just another case of one of the situations handled below.

Reply:
Sure.